use-latest-deps-node.sh

```diff
@@ -69,20 +69,30 @@ set -e
 # Install the npm-check-updates package so we can use `ncu`. It is insalled in
 # the parent directory to avoid polluting the repo's package.json file.
 npm --prefix ../ install npm-check-updates
+NCU=$(pwd)/../node_modules/.bin/ncu
 
 # Find all package.json files.
 files=$(find . -name "package.json" -not -path "**/node_modules/*")
 
 # Update dependencies in all package.json files.
 for file in $files; do
+  FILE_DIR=$(dirname "${file}")
+
+  # Move into the file's directory and run ncu, this auto-detects
+  # the location of the package.json file and any .ncurc files
   if [[ "$REGEX" == 0 ]]; then
-    ../node_modules/.bin/ncu -u -a --packageFile "${file}";
+    (
+    cd "${FILE_DIR}"
+    $NCU -u -a;
+    )
   else
-    ../node_modules/.bin/ncu -u -a -f "${REGEX}" --packageFile "${file}";
+    (
+    cd "${FILE_DIR}"
+    $NCU -u -a -f "${REGEX}";
+    )
   fi
 
   # If the folder contains a package-lock.json file then run `npm install` to also update the lock file.
```

Comment:
firebase-js-sdk uses yarn file. We probably will need to check which type of package manager is being used and run the corresponding command.

For monorepo, the yarn.lock is only present at the root folder, so in this case, we will need to run `yarn install` at the root folder if there is any update.

Reply:
Done!

Reply:
Unfortunately It still doesn't work for monorepo, we should run `yarn install` at the root of the repo once instead of in each directory.